src/Compilers/CSharp/Test/Symbol/Compilation/GetSemanticInfoTests.cs

```diff
@@ -5916,5 +5916,36 @@ public void GetSpecialType_ThrowsOnGreaterThanCount()
 
             Assert.True(exceptionThrown, $"{nameof(comp.GetSpecialType)} did not throw when it should have.");
         }
+
+        [Fact]
+        [WorkItem(34984, "https://github.com/dotnet/roslyn/issues/34984")]
+        public async Task ConversionIsExplicit_UnsetConversionKind()
+        {
+            var source =
+@"class C1
+{
+}
+
+class C2
+{
+    public void Foo() 
+    {
+        var c = new C1();
+        foreach (string item in c.Items)
+        {
+        }
+}";
+            var comp = CreateCompilation(source);
+            var tree = comp.SyntaxTrees.Single();
+            var model = comp.GetSemanticModel(tree);
+
+            var root = await tree.GetRootAsync();
+            var foreachSyntaxNode = root.DescendantNodes().OfType<ForEachStatementSyntax>().Single();
+            var foreachSymbolInfo = model.GetForEachStatementInfo(foreachSyntaxNode);
+
+            Assert.Equal(Conversion.UnsetConversion, foreachSymbolInfo.CurrentConversion);
```

Comment:
>UnsetConversion [](start = 36, length = 15)

Feels like there is a bigger issue here. I'd expect the conversion to be `Conversion.NoConversion` (and `Exists` to be `false`).

I think that was changed by PR [https://github.com/dotnet/roslyn/pull/33648](https://github.com/dotnet/roslyn/pull/33648) which introduced `UnsetConversion`, but was not intentional.
Let's discuss with @gafter

Reply:
I'm going to continue with this PR as is to fix the crash. I've filed #35918 to track a deeper fix if needed.